scripts/normalize-by-median.py

```diff
@@ -233,13 +236,17 @@ def main():  # pylint: disable=too-many-branches,too-many-statements
     discarded = 0
     input_filename = None
 
-    for index, input_filename in enumerate(args.input_filenames):
-        if args.single_output_filename != '':
-            output_name = args.single_output_filename
+    if args.single_output_filename != '':
+        output_name = args.single_output_filename
+        if args.append:
             outfp = open(args.single_output_filename, 'a')
         else:
+            outfp = open(args.single_output_filename, 'w+')
+
+    for index, input_filename in enumerate(args.input_filenames):
+        if not args.single_output_filename:
             output_name = os.path.basename(input_filename) + '.keep'
-            outfp = open(output_name, 'w')
+            outfp = open(output_name, 'w+')
```

Comment:
Line 248 overrides line 240
Line 249 overrides line 239 & 244.

Are there tests with multiple input files using `--out` with and without `--append`? There is likely a bug here. Regardless the output_name and outfp should be defined only once.